cmd/finch/nerdctl_windows.go

```diff
@@ -107,21 +106,250 @@ func handleVolume(systemDeps NerdctlCommandSystemDeps, args []string, index int)
+func handleOutputOption(systemDeps NerdctlCommandSystemDeps, args []string, index int) error {
+	var prefix = args[index]
+	var v = ""
+	var found = false
+	var before = ""
+	if strings.Contains(args[index], "=") {
+		before, v, found = strings.Cut(prefix, "=")
+	} else {
+		if (index + 1) < len(args) {
+			v = args[index+1]
+		} else {
+			return fmt.Errorf("invalid positional parameter for %s", prefix)
+		}
+	}
+
+	// https://docs.docker.com/engine/reference/commandline/build/ order does not matter, so convert to a map
+	entries := strings.Split(v, ",")
+	m := make(map[string]string)
+	for _, e := range entries {
+		parts := strings.Split(e, "=")
+		m[strings.TrimSpace(parts[0])] = strings.TrimSpace(parts[1])
+	}
+	dest, ok := m["dest"]
+	if !ok {
+		return nil
+	}
+	wslPath, err := convertToWSLPath(systemDeps, dest)
+	if err != nil {
+		return err
+	}
+	m["dest"] = wslPath
+
+	// Convert to string representation
+	s := mapToString(m)
+
+	if found {
+		args[index] = fmt.Sprintf("%s=%s", before, s)
+	} else {
+		args[index+1] = s
+	}
+
+	return nil
+}
+
+// handles --secret option for build command.
+func handleSecretOption(systemDeps NerdctlCommandSystemDeps, args []string, index int) error {
+	var prefix = args[index]
+	var v = ""
+	var found = false
+	var before = ""
+	if strings.Contains(args[index], "=") {
+		before, v, found = strings.Cut(prefix, "=")
+	} else {
+		if (index + 1) < len(args) {
+			v = args[index+1]
+		} else {
+			return fmt.Errorf("invalid positional parameter for %s", prefix)
+		}
+	}
+
+	entries := strings.Split(v, ",")
+	m := make(map[string]string)
+	for _, e := range entries {
+		parts := strings.Split(e, "=")
+		m[strings.TrimSpace(parts[0])] = strings.TrimSpace(parts[1])
+	}
+	sp, ok := m["src"]
+	if !ok {
+		return nil
+	}
+	wslPath, err := convertToWSLPath(systemDeps, sp)
+	if err != nil {
+		return err
+	}
+	m["src"] = wslPath
+
+	// Convert to string representation
+	s := mapToString(m)
+
+	if found {
+		args[index] = fmt.Sprintf("%s=%s", before, s)
+	} else {
+		args[index+1] = s
+	}
+
+	return nil
+}
+
+// cp command handler, takes command arguments and converts hostpath to wsl path in place. It ignores all other arguments
+func cpHandler(systemDeps NerdctlCommandSystemDeps, cmdArgs []string) error {
+	for i, arg := range cmdArgs {
+		// -L and --follow-symlink don't have to be processed
+		if arg == "-" || strings.HasPrefix(arg, "-") {
+			continue
+		} else {
+			// If argument contains container path, then continue
+			colon := strings.Index(arg, ":")
+
+			// this is a container path
+			if colon > 1 {
+				continue
+			}
+			wslPath, err := convertToWSLPath(systemDeps, arg)
+			if err != nil {
+				return err
+			}
+			cmdArgs[i] = wslPath
+		}
+	}
+	return nil
+}
+
+// this is the handler for image build command. It translates build context to wsl path.
+func imageBuildHandler(systemDeps NerdctlCommandSystemDeps, cmdArgs []string) error {
+	var err error
+	if cmdArgs[len(cmdArgs)-1] != "--debug" {
```

Comment:
Can only the first option be `--debug`? Do we have to search arguments instead of assuming order, or is this a safe assumption?

Reply:
It does not matter, if the last argument is `--debug`, then build context is the one before that, if not the last argument is buildcontext. `--debug` will be skipped from processing either way. It's mainly to check if --debug is the last argument while translating buildcontext.